plugin-feature-doc.md

```diff
@@ -0,0 +1,111 @@
+# Plugin API is simple and well-documented
+
+The current plugin API has an overly large surface area, poor documentation,
+and unclear names.  The objective is to improve this, focusing on the UX of
+primary use-case: creating new service authenticators.
+
+Aha Card: https://cyberark.aha.io/features/AAM-92
+
+- [Objective](#objective)
+- [Experience (Summary)](#experience-summary)
+- [Experience (Detailed)](#experience-detailed)
+- [Technical Details](#technical-details)
+- [Testing](#testing)
+- [Documentation](#documentation)
+- [Open Questions](#open-questions)
+- [Stories](#stories)
+- [Future Work](#future-work)
+
+### Objective
+
+- Simplify Plugin API so authors of new service authenticators have a good
+  experience which requires learning only concepts relevant to their goal.
+- Write clear documentation to make the simplified API accessible to new
+  contributors
+
+### Experience (Summary)
+
+The new plugin API will support writing new service authenticators only.  Only
+a minimal interface will be required.
+
+### Experience (Detailed)
+
+Authors will need to implement only `PluginInfo` and a new
+`ServiceAuthenticator` interface.
+
+`APIVersion` will be moved into `PluginInfo`, and the `plugin_type` will be
+explicitly specified:
+
+```go
+var PluginInfo = map[string]string{
+  "api_version": "2.0",
+  "plugin_types": "service_authenticator",
+  "version":     "0.0.7",
+  "id":          "test-plugin",
+  "name":        "Test Plugin",
+  "description": "Test plugin to demonstrate plugin functionality",
+}
+```
+
+Implementing authenticators will look like:
+
+```go
+func ServiceAuthenticators() map[string]func(plugin_v2.AuthenticatorOptions) plugin_v2.Authenticator {
```

Comment:
this might be a "too long since I've been in the code" question, but is this what will replace [this part](https://github.com/cyberark/secretless-broker/blob/master/test/plugin/example/cmd/main.go#L27) of the example plugin?

can we break this out in the design doc like we did for the config, where we said "here is what the example plugin looked like before" and "here is what we propose to look like it now"? I feel like that would make the improvement clearer and would give us an opportunity to also address how we are dropping support for other plugin types for now.

Reply:
> this might be a "too long since I've been in the code" question, but is this what will replace this part of the example plugin?

Yep.

> can we break this out in the design doc like we did for the config, where we said "here is what the example plugin looked like before" and "here is what we propose to look like it now"?

Yep, but to be clear as of now this one is a simple rename.  So there's no improvement beyond that.  And  it looks like that name will actually be `ServiceConnecters()` or `Connecters()`

Reply:
just a heads up that it should be `Connectors` and not `Connecters`.